Add manifest for KDE Kile

Pointedstick opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions.
  - `--filesystem=home` is to make the embedded terminal not be useless
  - `--env=PATH=/app/bin:/app/texlive/bin:/app/texlive/bin/x86_64-linux:/usr/bin/` is to make Tex stuff work
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
  - https://invent.kde.org/ngraham
  - https://invent.kde.org/office/kile/-/commits/master?author=Nate%20Graham
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.
  - https://invent.kde.org/office/kile/-/commit/8a06742bdf65c169c9a015cba21b78f2a1fbb508
  - https://invent.kde.org/office/kile/-/commit/01c0c26c065c2ddabe8d87b0330ed0363c3e4bed

---

This app includes functionality from KDE's Konsole and Okular apps, so the packaging copies Okular's pre-existing FlatHub packaging to bundle dependencies not already in a runtime. cc @tsdgeos

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

tsdgeos commented:
bot, build org.kde.kile

Pointedstick commented:
Thanks for the suggestions! I will look into those shortly.

Pointedstick commented on org.kde.kile.json:
```diff
@@ -0,0 +1,408 @@
+        "--socket=cups",
+        "--socket=x11",
+        "--socket=wayland",
+        "--socket=session-bus",
```
This may not be needed after all; I will see if I can remove it.

hfiguiere commented:
AppStream file doesn't validate:

```
builddir/files/share/appdata/org.kde.kile.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <release> required
Validation of files failed
```

Pointedstick commented:
Fixed that upstream with https://invent.kde.org/office/kile/-/commit/8a06742bdf65c169c9a015cba21b78f2a1fbb508, and added a patch to the packaging for now until we can update to the next version.

hfiguiere commented:
bot, build org.kde.kile

hfiguiere commented:
```
builddir/files/share/appdata/org.kde.kile.appdata.xml: FAILED:
• tag-missing           : <release> required
Validation of files failed
```

Pointedstick commented:
bot, build org.kde.kile

nedrichards commented:
/merge
